src/pheval/analyse/analysis.py

```diff
@@ -105,44 +45,60 @@ def benchmark_directory(
     disease_analysis: bool,
     plot_type: str,
 ) -> None:
-    """Benchmark prioritisation performance for a single directory."""
-    gene_stats_writer = (
-        RankStatsWriter(Path(output_prefix + "-gene_summary.tsv")) if gene_analysis else None
-    )
-    variants_stats_writer = (
-        RankStatsWriter(Path(output_prefix + "-variant_summary.tsv")) if variant_analysis else None
-    )
-    disease_stats_writer = (
-        RankStatsWriter(Path(output_prefix + "-disease_summary.tsv")) if disease_analysis else None
-    )
-    gene_rank_comparison, variant_rank_comparison, disease_rank_comparison = (
-        defaultdict(dict),
-        defaultdict(dict),
-        defaultdict(dict),
-    )
-    prioritisation_data = _assess_prioritisation_for_results_directory(
-        results_dir_and_input,
-        score_order,
-        threshold,
-        gene_rank_comparison,
-        variant_rank_comparison,
-        disease_rank_comparison,
-        gene_stats_writer,
-        variants_stats_writer,
-        disease_stats_writer,
-        gene_analysis,
-        variant_analysis,
-        disease_analysis,
+    """Benchmark prioritisation performance for a result directory."""
+    if gene_analysis:
+        run_benchmark(
+            results_dir_and_input=results_dir_and_input,
+            score_order=score_order,
+            output_prefix=output_prefix,
+            threshold=threshold,
+            plot_type=plot_type,
+            benchmark_generator=GeneBenchmarkRunOutputGenerator(),
+        )
+    if variant_analysis:
+        run_benchmark(
+            results_dir_and_input=results_dir_and_input,
+            score_order=score_order,
+            output_prefix=output_prefix,
+            threshold=threshold,
+            plot_type=plot_type,
+            benchmark_generator=VariantBenchmarkRunOutputGenerator(),
+        )
+    if disease_analysis:
+        run_benchmark(
+            results_dir_and_input=results_dir_and_input,
+            score_order=score_order,
+            output_prefix=output_prefix,
+            threshold=threshold,
+            plot_type=plot_type,
+            benchmark_generator=DiseaseBenchmarkRunOutputGenerator(),
+        )
+
+
+def run_benchmark_comparison(
```

Comment:
Should `run_benchmark_comparison` and `run_benchmark` be a 'private' method? i.e. `_run_benchmark_comparison` and `_run_benchmark` as the names `benchmark` and `benchmark_run_comparisons` are somewhat ambiguous when compared with `run_benchmark_comparison` and `run_benchmark`  and it's not immediately clear which methods should be being called externally.